### ISSUE TITLE
URLs listed in Always Purge URL(s) clear from WP Rocket locally, but not from Varnish

### ISSUE THREAD

**Issue by charlescwcooke** [wp-media/wp-rocket#4441]
**Describe the bug**
We have WP Rocket installed. The Varnish add-on is switched on. Because the Varnish server isn't running on localhost, we are also using the `wp-rocket-compat-varnish-ip` to specify a custom IP for the Varnish server. This works well. When we clear the WP Rocket cache — whether that be the whole cache, or just updating a page — it also clears the cache on Varnish as expected.

But we have recently noticed a bug — or, at least, an exception to this. Some of our custom pages are not natively cleared by WP Rocket, or by Varnish, presumably because it doesn't know about them. To fix this, we've added their URLs into the *Always Purge URL(s)* field in *Advanced Rules*. Doing this does prompt WP Rocket to clear those pages from its cache upon an update, but, for some reason, it doesn't translate through to Varnish. So, while those pages are cleared on the local server, they're not cleared on the Varnish server that's sitting in front of it.

TL;DR: Using WP Rocket with Varnish and the `wp-rocket-compat-varnish-ip` add-on, URLs placed into the *Always Purge URL(s)* field clear on the local server but not on Varnish.

**Expected behavior**
I expected the URLs listed in the *Always Purge URL(s)* field to clear from Varnish as well as WP Rocket when a post or page is updated.

**Comment by sarangsss29** [comment]
Bump. Want to know about this too.

**Comment by piotrbak** [comment]
Hello @charlestroluxe 

Varnish is clearing exactly the same files as WP Rocket is doing:
https://github.com/wp-media/wp-rocket/blob/aea8f96b0725f900deaa337af2f87c37277676ff/inc/Addon/Varnish/Subscriber.php#L44

https://github.com/wp-media/wp-rocket/blob/aea8f96b0725f900deaa337af2f87c37277676ff/inc/common/purge.php#L103

However, I can see that URLs passed from this field to the Varnish look like this `http://domain.com//varnishpurge/.*`, not sure whether the double slash might be a problem. I'd suggest using this filter in order to add the correct URL to the purge list here:
https://github.com/wp-media/wp-rocket/blob/aea8f96b0725f900deaa337af2f87c37277676ff/inc/common/purge.php#L306

Or hardcoding the correct URL for testing purposes here:
https://github.com/wp-media/wp-rocket/blob/aea8f96b0725f900deaa337af2f87c37277676ff/inc/Addon/Varnish/Varnish.php#L18

If that won't help, please reach our support for further investigation.
